flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/SnapshotUtils.java

```diff
@@ -264,101 +264,103 @@ static boolean shouldTriggerAutomaticSnapshot(
             return false;
         } // automaticTriggerExpression was configured by the user
 
-        // Try to interpret as an interval (Duration).
-        boolean expressionIsAnInterval = true;
-        boolean shouldTriggerInterval = false;
-        try {
-            shouldTriggerInterval =
-                    shouldTriggerIntervalBasedSnapshot(
-                            snapshotType, automaticTriggerExpression, lastTrigger);
-        } catch (ParseException e) {
-            expressionIsAnInterval = false;
-        }
-
-        // Try to interpret as a cron expression.
-        boolean expressionIsACron = true;
-        boolean shouldTriggerCron = false;
-        try {
-            shouldTriggerCron =
-                    shouldTriggerCronBasedSnapshot(
-                            snapshotType, automaticTriggerExpression, lastTrigger, Instant.now());
-        } catch (ParseException e) {
-            expressionIsACron = false;
-        }
-
-        if (!expressionIsAnInterval && !expressionIsACron) {
-            LOG.warn(
-                    "Automatic {} triggering is configured, but the trigger expression '{}' is neither a valid Duration, nor a cron expression.",
-                    snapshotType,
-                    automaticTriggerExpression);
-            return false;
-        }
+        boolean isInterval = isInterval(automaticTriggerExpression);
+        boolean isCron = isCron(automaticTriggerExpression);
 
         // This should never happen. The string cannot be both a valid Duration and a cron
         // expression at the same time.
-        if (expressionIsAnInterval && expressionIsACron) {
+        if (isInterval && isCron) {
             LOG.error(
                     "Something went wrong with the automatic {} trigger expression {}. This setting cannot be simultaneously a valid Duration and a cron expression.",
                     snapshotType,
                     automaticTriggerExpression);
             return false;
```

Comment:
The question is whether we want to skip triggering entirely or we choose to go with the interval. I think semantically it is not possible for a proper cron to be an interval, so this is is an extreme edge case. Also we fail silently and continue operation which is also debatable. 

I would just go with whatever matches first, i.e. the interval, but maybe there is no right and wrong.

Reply:
I think if there is any such weird case possible, I'd prefer to uncover it. If a user configures an interval, it is reasonable tot expect that she would eventually check it is is firing. In that case t is arguably better not to fire at all and potentially get a bug submitted than trigger a random thing that matches, but is interpreted in the wrong way.